src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,605 @@
+                raise Exception("no friendly robots found on field")
+
+        except queue.Empty as empty:
+            raise Exception("unable to determine robots on the field")
+
+    def send_gamecontroller_command(
+        self,
+        gc_command: proto.ssl_gc_state_pb2.Command,
+        team: proto.ssl_gc_common_pb2.Team,
+        final_ball_placement_point=None,
+    ):
+
+        self.gamecontroller.send_ci_input(
+            gc_command=gc_command,
+            team=team,
+            final_ball_placement_point=final_ball_placement_point,
+        )
+
+    def set_worldState(self, worldstate: WorldState):
+        """ Sets a world state on the field by moving robots and the ball to the given positions
+
+        Args:
+            worldstate (WorldState): Proto which contains robot and ball positions expected on the field
+
+        Raises:
+            Exception: On failing to verify that the expected position was set
+        """
+
+        self.__validateMatchingRobotIds(worldstate)
+        self.__setBallState(worldstate)
+        self.__setRobotState(worldstate)
+
+    def __validateWorldState(self, eventually_validation_set, timeout):
+        timeout_time = time.time() + timeout
+
+        while time.time() < timeout_time:
+            try:
+
+                current_world = self.world_buffer.get(
+                    block=True, timeout=WORLD_BUFFER_TIMEOUT
+                )
+
+                (
+                    eventually_validation_status,
+                    always_validation_status,
+                ) = validation.run_validation_sequence_sets(
+                    world=current_world,
+                    eventually_validation_sequence_set=[[eventually_validation_set]],
+                    always_validation_sequence_set=[[]],
+                )
+
+                if not validation.contains_failure(eventually_validation_status):
+                    break
+
+            except queue.Empty as empty:
+                logger.warning("failed to obtain world")
+
+        validation.check_validation(eventually_validation_status)
+
+    def __validateMatchingRobotIds(self, worldstate: WorldState):
+        """Checks whether the robots to be used in this field test matches up with the robots we see on the field
+
+        Args:
+            worldstate (worldState Proto): The worldstate containing the robots to be used in the test
+
+        Raises:
+            Exception: On there existing a robot to be used in the test but is not on the field
+        """
+        logger.info("testing ids matching")
+
+        # validate that ids match with test setup
+        ids_present = True
+        for idx, robot_ids in enumerate(
+            [worldstate.blue_robots, worldstate.yellow_robots]
+        ):
+            team_name = "friendly" if idx == 0 else "enemy"
+            team_robots = (
+                self.friendly_robot_ids_field
+                if idx == 0
+                else self.enemy_robot_ids_field
+            )
```

Comment:
If we end up adding a cmdline arg for selecting friendly color, then this logic should also be updated.

Reply:
see [comment below](https://github.com/UBC-Thunderbots/Software/pull/2906/files/4da5b4bad0f98aac1dd0091306efd6fe0144c046#r1194576241) about having 2 full systems

Reply:
This might have to be updated given the recent changes with only 1 fullsystem. Seems like it always treats blue as friendly.